components/data-provider/src/structs/dates.rs

```diff
@@ -0,0 +1,131 @@
+// Date
+#[cfg(feature = "invariant")]
+use crate::prelude::*;
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+#[cfg(feature = "invariant")]
+pub(crate) fn get_invariant(data_key: &DataKey) -> Option<DataResponse<'static>> {
+    use crate::invariant::make_inv_response;
+    if data_key.category != DataCategory::Dates {
+        return None;
+    }
+    match (data_key.sub_category.as_str(), data_key.version) {
+        ("gregorian", 1) => make_inv_response::<gregorian::DatesV1>(),
+        _ => None,
+    }
+}
+
+pub mod gregorian {
+    use serde::{Deserialize, Serialize};
+    use std::borrow::Cow;
+    #[derive(Debug, PartialEq, Clone, Deserialize, Serialize, Default)]
+    pub struct DatesV1 {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub months: Option<months::ContextsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub days: Option<days::ContextsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub day_periods: Option<day_periods::ContextsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub date_style_patterns: Option<patterns::StylePatternsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub time_style_patterns: Option<patterns::StylePatternsV1>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub date_time_style_patterns: Option<patterns::StylePatternsV1>,
```

Comment:
I flattened it here, but I'm tempted to add a StylePatternsV1 struct with date, time and date_time with just styles. What do you think? @mihnita @sffc ?